dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/UnresolvableArtifactProblem.java

```diff
@@ -41,6 +42,18 @@ public UnresolvableArtifactProblem(Artifact artifact) {
   public UnresolvableArtifactProblem(List<DependencyNode> dependencyPath) {
     super(dependencyPath.get(dependencyPath.size() - 1).getArtifact(), dependencyPath);
   }
+  
+  /**
+   * Problem when Maven cannot resolve the artifact at the leaf of {@code dependencyPath} in a
+   * dependency tree.
+   * 
+   * @param artifact dependency that can't be found
+   * @param path to the unfound artifact, not including the unfound artifact
```

Comment:
`path` needs to include the unfound artifact; otherwise we cannot tell the scope / optional of the dependency.

Reply:
Let me double check, but I think that would be a behavioral change from the current code

Reply:
OK, I think you're right though the code is a little hard to follow. The parentNodes stack variable might be poorly named. In any case I can make this doc comment ambiguous enough to cover both cases, and this PR does not change the behavior

Reply:
The parentNodes do not have the node being checked. ExceptionAndPath was combining them to make the dependency path for UnresolvableArtifactProblem.

Reply:
> The parentNodes do not have the node being checked.

This was wrong.